lib/statesman/machine.rb

```diff
@@ -287,7 +292,12 @@ def validate_transition(options = { from: nil, to: nil, metadata: nil })
 
       # Call all guards, they raise exceptions if they fail
       guards_for(from: from, to: to).each do |guard|
-        guard.call(@object, last_transition, options[:metadata])
+        guard.call(
+          @object,
+          last_transition,
+          options[:metadata],
+          from: from, to: to,
```

Comment:
I'm worried that this violates the upcoming change to kwargs (https://bugs.ruby-lang.org/issues/14183) 

While the change to error message is helpful since we don't currently provide details about `to` / `from` on the error object itself people may be parsing the message and changing that would be breaking.

Reply:
> While the change to the error message is helpful since we don't currently provide details about to / from on the error object itself people may be parsing the message and changing that would be breaking.

That's a very good point, Dan -- so if I understand your ask correctly, I should limit this PR to just points 1 and 2 above, and remove the changes which pass in transition information to the callback?

Reply:
Yeah. I'll have a think about how to pass the information down and once I've figure that out I'll release a version bump that changes that message too.

Reply:
Sounds good 👍